smrf/distribute/wind/wind.py

```diff
@@ -69,41 +70,52 @@ class Wind(image_data.image_data):
     ])
 
     def __init__(self, config):
-
-        # extend the base class
         image_data.image_data.__init__(self, self.VARIABLE)
         self._logger = logging.getLogger(__name__)
 
         # check and assign the configuration
         self.smrf_config = config
         self.getConfig(config['wind'])
 
-        if self.check_wind_model_type('interp'):
+        if self.model_type(self.INTERP):
             # Straight interpolation of the wind
             self.wind_model = self
             self.wind_model.flatwind = None
             self.wind_model.cellmaxus = None
             self.wind_model.dir_round_cell = None
 
-        elif self.check_wind_model_type('wind_ninja'):
+        elif self.model_type(WindNinjaModel.MODEL_TYPE):
             self.wind_model = WindNinjaModel(self.smrf_config)
 
-        elif self.check_wind_model_type('winstral'):
+        elif self.model_type(WinstralWindModel.MODEL_TYPE):
             self.wind_model = WinstralWindModel(self.smrf_config)
 
         self._logger.debug('Created distribute.wind')
 
-    def check_wind_model_type(self, wind_model):
-        """Check if the wind model is of a given type
+    def model_type(self, wind_model):
+        """Check if given model is set on config
+
+        Args:
+            wind_model (str): name of the wind model
+
+        Returns:
+            bool: True/False
+        """
+        return Wind.config_model_type(self.smrf_config, wind_model)
+
+    @staticmethod
+    def config_model_type(config, wind_model):
+        """Check if the wind model is of a given type for given config
 
         Args:
+            config (dict): run configuration for SMRF
             wind_model (str): name of the wind model
 
         Returns:
-            bool: True/False if the wind_model matches the config
+            bool: True/False if the wind_model is set in the config
         """
 
-        return self.config['wind_model'] == wind_model
+        return config.get('wind', {}).get('wind_model', None) == wind_model
```

Comment:
Is the `.get('wind', {})` needed? inicheck will complain if the wind section is not in the config so it's safe to assume that `wind` will be there.

Reply:
This will ensure that the call won't crash if you were to use SMRF outside of an initialized run via a `.ini`. 
It's something I adapted over the years, to never assume a root key exists when checking for a child key.
The existence check for a key is easy to do and does not hurt performance